polaris-react/src/components/IndexTable/IndexTable.tsx

```diff
@@ -455,6 +465,7 @@ function IndexTableBase({
     resizeTableHeadings,
     firstStickyHeaderElement,
     tableInitialized,
+    stickyHeaderInitialized,
```

Comment:
Is the appropriate formatting of the sticky header dependent on this `useEffect` firing, which is why we encountered the issue on initial render as the sticky header element was not mounted yet? However, why would this work when navigating away and coming back, which would still be a remount?

Without fully understanding this logic, it does seem like this pattern existed for `tableInitialized` and maybe this was a miss with the sticky header?

Reply:
Good point. Adding just that didn't seem to fix the issue, so likely no. Will remove and retest 👍